app/controllers/posts_controller.rb

```diff
@@ -2,12 +2,31 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
     @posts = @user.posts
-    render :index
+    redirect_to :root, alert: 'Warning: No posts' unless @posts.length.positive?
   end
 
   def show
     @post = Post.find(params[:id])
     @comments = @post.comments
     render :show
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
```

Comment:
- [ ] **Optional:** When you create action/methods, always create room for errors when they're executed. This is implemented by creating try and catch with if ... else.

```
 if (#it all goes well) 
     # run some code
 else
     # run some code
 end
```